### PR TITLE
Only make a shallow copy when loading optimizer state_dict

### PR DESCRIPTION
**Pull request by janeyx99**
The thing we do still deep copy is the param_groups, which is much lighter weight. This should also save memory when loading from a checkpoint.

The deepcopy was introduced in https://github.com/pytorch/pytorch/commit/ecfcf39f302f7bb193884f72ef2bd59141e5c46c, but module.py had only a shallow copy at that point so it did not actually bring parity.

Incorporates an XLA fix, which is why I'm updating the pin to https://github.com/pytorch/xla/commit/ca5eab87a71f80cd3168630511d02549cc7d2516

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #106082

### CODE REVIEW

File: .git-blame-ignore-revs [hunk 1]
```diff
@@ -30,3 +30,5 @@ e3900d2ba5c9f91a24a9ce34520794c8366d5c54
 30fb2c4abaaaa966999eab11674f25b18460e609
 # 2023-06-06 clang-format on Foreach / Multi-Tensor-Apply
 515c4279416f13fcc3c898e560f8ae8f15139a03
+# [optim][BE] split test file into logical parts: SWA, LR, optim
+a53cda1ddc15336dc1ff0ce1eff2a49cdc5f882e
```
Comment (janeyx99) [inline .git-blame-ignore-revs:34]: This could go in another PR but it didn't feel worth another PR....I can be easily convinced otherwise

Comment (albanD) [reply .git-blame-ignore-revs:34]: TIL this file exists and github uses it by default. Very cool!

**Review by albanD — APPROVED** [review]
SGTM! Very good catch!

**Review by Skylion007 — APPROVED** [review]

**Review by albanD — APPROVED** [review]
Ok!

**Pull request closed without merging**